csharp/src/Ice/Proxy.cs

```diff
@@ -302,6 +302,7 @@ public static Connection GetConnection(this IObjectPrx prx)
                 bool releaseRequestAfterSent =
                     requestSize > reference.Communicator.RetryRequestSizeMax ||
                     !reference.Communicator.IncRetryBufferSize(requestSize);
+                bool clearedLocatorCache = false;
```

Comment:
I see your point, I think a good solution here will be to tell GetEndpointsAsync to get fresh endpoints without clearing the cache first, only the call that gets the exception will wait for the resolution, once the endpoint resolution response arrives we swap the cache, and concurrent calls will start using the new endpoints.

The other optimization is that this will be only done if the endpoints were cached in the first place, no point in clearing the cache if the first call already got fresh endpoints.

regarding OtherReplica I think you are right, no point in refresh the endpoints if the current endpoint is still valid for retries.